snuba/query/snql/expression_visitor.py

```diff
@@ -0,0 +1,196 @@
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    List,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+from parsimonious import Node
+from snuba.query.dsl import divide, minus, multiply, plus
+from snuba.query.expressions import (
+    Column,
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    Literal,
+)
+
+
+class LowPriOperator(Enum):
+    PLUS = "+"
+    MINUS = "-"
+
+
+class HighPriOperator(Enum):
+    MULTIPLY = "*"
+    DIVIDE = "/"
+
+
+class LowPriTuple(NamedTuple):
+    op: LowPriOperator
+    arithm: Expression
+
+
+class HighPriTuple(NamedTuple):
+    op: HighPriOperator
+    arithm: Expression
+
+
+HighPriArithmetic = Union[Node, HighPriTuple, Sequence[HighPriTuple]]
+LowPriArithmetic = Union[Node, LowPriTuple, Sequence[LowPriTuple]]
+
+
+def get_arithmetic_function(
+    operator: Enum,
+) -> Callable[[Expression, Expression, Optional[str]], FunctionCall]:
+    return {
+        HighPriOperator.MULTIPLY: multiply,
+        HighPriOperator.DIVIDE: divide,
+        LowPriOperator.PLUS: plus,
+        LowPriOperator.MINUS: minus,
+    }[operator]
+
+
+def get_arithmetic_expression(
+    term: Expression, exp: Union[LowPriArithmetic, HighPriArithmetic],
+) -> Expression:
+    if isinstance(exp, Node):
+        return term
+    if isinstance(exp, (LowPriTuple, HighPriTuple)):
+        return get_arithmetic_function(exp.op)(term, exp.arithm, None)
+    elif isinstance(exp, list):
+        for elem in exp:
+            term = get_arithmetic_function(elem.op)(term, elem.arithm, None)
+    return term
+
+
+class ExpressionVisitor:
+    def visit_function_name(left, node: Node, visited_children: Iterable[Any]) -> str:
```

Comment:
Shouldn't these methods be used by the old expression parser as well ? https://github.com/getsentry/snuba/blob/master/snuba/query/parser/expressions.py#L117

Or do you plan to make that change later ?

Reply:
Yes, I think I can fix the init file first and then apply it to Clickhouse Visitor.